monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py

```diff
@@ -54,3 +54,18 @@ def test_island_mode_post__set_invalid_model(flask_client, uses_database):
         "/api/island-mode", data=json.dumps({"mode": "bogus mode"}), follow_redirects=True
     )
     assert len(IslandMode.objects) == 0
+
+
+@pytest.mark.parametrize("mode", ["ransomware", "advanced"])
+def test_island_mode_get(flask_client, uses_database, mode):
```

Comment:
this also tests the `post`. Maybe we don't need the `test_island_mode_post__set_model` and `test_island_mode_post__set_invalid_model` tests anymore?

Reply:
Tests should be individual i.e. testing one kind of behaviour. You should set `IslandMode.objects` and then test just the post request.

Reply:
There's no need to make unit tests too granular. The result of the `post` is that `get` returns the mode posted. There's no need to tie the unit test to the mechanism of how the mode is saved. If we decide to switch the database we won't need to re-do unit tests, for example.

Reply:
I don't agree. If the test fails, you're now left debugging more than if the tests were granular. 
I don't think our motive behind writing unit tests should be to not have to redo them ever again. It should be to make debugging and catching issues easier.